tests/scripts/rm_hpo_api_tests.sh

```diff
@@ -328,8 +329,245 @@ function other_post_experiment_tests() {
+	for exp_trial in "${get_trial_json_invalid_tests[@]}"
+	do
+		TESTS_="${TEST_DIR}/${exp_trial}"
+		mkdir -p ${TESTS_}
+		LOG_="${TEST_DIR}/${exp_trial}.log"
+		result="${TESTS_}/${exp_trial}_result.log"
+		echo "************************************* ${exp_trial} Test ****************************************" | tee -a ${LOG_} ${LOG}
+		
+		# Start the HPO servers
+		${SCRIPTS_DIR}/start_hpo_servers.sh -p ${TESTS_} | tee -a ${LOG_} ${LOG}
+		
+		# Sleep for few seconds to reduce the ambiguity
+		sleep 2
+		
+		# Check if the servers have started
+		check_server_status
+
+		# Get the id from search space JSON
+		current_id=$(cat ${SEARCH_SPACE_JSON} | jq .[].id | tr -d '""')
+
+		create_post_exp_json_array ${current_id}
+		post_experiment_json "${rm_hpo_post_experiment_json[$experiment]}"
+
+		run_exp_trial_test ${exp_trial}
+
+		actual_result="${http_code}"
+		
+		expected_result_="^4[0-9][0-9]"
+		expected_behaviour="RESPONSE_CODE = 4XX BAD REQUEST"
+
+		compare_result ${__test_name__} ${expected_result_} "${expected_behaviour}"
+		echo ""
+		
+		# Stop the HPO servers
+		${SCRIPTS_DIR}/start_hpo_servers.sh -t | tee -a ${LOG_} ${LOG}
+	done
+	echo "*********************************************************************************************************" | tee -a ${LOG_} ${LOG}
+}
+
+# Validate if the actual tunable name is matching with the tunable name returned by dependency analyzer
+function validate_tunable_name() {
+	failed=0
+	if [ "${actual_tunable_name}" != "${tunable_name}" ]; then
+		failed=1
+	fi
+	expected_behaviour="Actual Tunable name should match with the tunable name returned by dependency analyzer"
+	display_result "${expected_behaviour}" ${FUNCNAME} ${failed}
+}
+
+# Validate if Actual Tunable value is within the given range
+function validate_tunable_value(){
+	failed=0
+	
+ 	if [[ $(bc <<< "${actual_tunable_value} >= ${lowerbound} && ${actual_tunable_value} <= ${upperbound}") == 0 ]]; then
+ 		failed=1
+	fi
+	expected_behaviour="Actual Tunable value should be within the given range"
+	display_result "${expected_behaviour}" ${FUNCNAME} ${failed}
+}
+
+# Validate the trial json returned by RM-HPO GET operation
+function validate_exp_trial() {
+	# Sort the actual json based on tunable name
+	echo "$(cat ${result} | jq  'sort_by(.tunable_name)')" > ${result}
+
+	# Sort the json based on tunable name
+	echo "$(jq '[.[].tunables[] | {lower_bound: .lower_bound, name: .name, upper_bound: .upper_bound}] | sort_by(.name)' ${SEARCH_SPACE_JSON})" > ${parse_json}
+
+	expected_tunables=$(cat ${parse_json} | jq '. | length')
+	actual_tunables=$(cat ${result}  | jq '. | length')
+	
+	echo "___________________________________ Validate experiment trial __________________________________________"
+	
+	if [ "${expected_tunables}" -ne "${actual_tunables}" ]; then
+		failed=1
+	else
+		while [ "${tunable_count}" -lt "${expected_tunables}" ]
+		do
+			upperbound=$(cat ${parse_json} | jq '.['${tunable_count}'].upper_bound' | tr -d '""' | tr -d 'M')
```

Comment:
Check if you need to strip quotes or M. Would be good to check the latest search space json API